cpp/inc/bond/protocol/detail/rapidjson_helper.h

```diff
@@ -13,9 +13,34 @@
 #include <boost/call_traits.hpp>
 #include <boost/noncopyable.hpp>
 #include <boost/locale.hpp>
+
 #include "rapidjson/rapidjson.h"
 #include "rapidjson/error/en.h"
+
+// rapidjspn/document.h v1.1 uses std::min/max in ways that conflict
+// with macros defined in windows. This works around the issue.
+#ifdef _MSC_VER
+#if  defined(_WINDEF_) || defined(_MINWINDEF_)
+#ifndef NOMINMAX
+#pragma push_macro("min")
```

Comment:
Minor: can we indent the behavior (the pushes and the undefs) from the conditions?